lib/spack/spack/util/spack_yaml.py

```diff
@@ -239,6 +176,21 @@ def ignore_aliases(self, _data):
 OrderedLineDumper.add_representer(syaml_int, OrderedLineDumper.represent_int)
 
 
+# Allow syaml_dict objects to be represented by ruamel.yaml.dump. With this,
+# syaml_dict allows a user to provide an ordered dictionary to yaml.dump when
+# the RoundTripDumper is used.
+RoundTripDumper.add_representer(syaml_dict, RoundTripDumper.represent_dict)
+
+
+#: Max integer helps avoid passing too large a value to cyaml.
+maxint = 2 ** (ctypes.sizeof(ctypes.c_int) * 8 - 1) - 1
+
+
+def dump_noconfig(obj, default_flow_style, stream=None):
```

Comment:
what's the "noconfig" suffix here?  Does that just mean you're not writing a config file?  I can't think of a better name off the top of my head, but i had to think about this one -- is there a better name we could come up with?

Reply:
I think https://github.com/spack/spack/pull/11602#discussion_r326350318 will answer this.